src/data-access/services/basic-ftp-ftp-client.ts

```diff
@@ -0,0 +1,89 @@
+import { FtpClient } from '../interfaces/ftp-client'
+import { FtpConnectionConfiguration } from '../value-objects/ftp-connection-configuration'
+import * as basicFtp from 'basic-ftp'
+import { PassThrough } from 'node:stream'
+import { Logger } from '../../logger/logger'
+import { FileMetadata } from '../value-objects/file-metadata'
+import { ConnectionStatus } from '../enums/connection-status'
+
+const CONNECTION_TIMEOUT_DURATION_IN_MS: number = 4000
+
+export class BasicFtpFtpClient implements FtpClient {
+  private onConnectionStatusChangedCallback?: (connectionStatus: ConnectionStatus) => void
+  private readonly ftpClient: basicFtp.Client = new basicFtp.Client(CONNECTION_TIMEOUT_DURATION_IN_MS)
+  private readonly logger: Logger
+
+  public constructor(
+    private readonly configuration: FtpConnectionConfiguration,
+    logger: Logger,
+  ) {
+    this.logger = logger.tag(this.constructor.name)
+  }
```

Comment:
Just a formatting suggestion :)

```suggestion
  public constructor(private readonly configuration: FtpConnectionConfiguration, logger: Logger) {
    this.logger = logger.tag(this.constructor.name)
  }
```